crypto/nextgen_crypto/src/ed25519.rs

```diff
@@ -209,6 +209,14 @@ impl Genesis for Ed25519PrivateKey {
     }
 }
 
+impl NilSigningKey for Ed25519PrivateKey {
+    fn nil_signing_key() -> Self {
+        let mut buf = [0u8; ed25519_dalek::SECRET_KEY_LENGTH];
```

Comment:
Rather than using `ed25519_dalek::SECRET_KEY_LENGTH` could you expose a `length()` parameter in the `PrivateKey` trait similarly to what's in `PublicKey`:

https://github.com/libra/libra/blob/3744c8b466c02b1f2484acd1d521e54de7fcc193/crypto/nextgen_crypto/src/ed25519.rs#L227-L231